README.md

```diff
@@ -1,158 +1,339 @@
-* Update the `Memcached` CR status with the names of the memcached pods
+```Go
+type MemcachedSpec struct {
+    // Size is the size of the memcached deployment
+    Size int32 `json:"size"`
+}
+type MemcachedStatus struct {
+    // Nodes are the names of the memcached pods
+    Nodes []string `json:"nodes"`
+}
+```
+
+After modifying the `*_types.go` file always run the following command to update the generated code for that resource type:
 
-#### Watch for the Memcached custom resource definition
+```sh
+$ operator-sdk generate k8s
+```
 
-By default, the memcached-operator watches `Memcached` resource events as shown in `cmd/memcached-operator/main.go`.
+### Manager
+
+The main program for the operator `cmd/manager/main.go` initializes and runs the [Manager][manager_go_doc].
+
+The Manager will automatically register the scheme for all custom resources defined under `pkg/apis/...` and run all controllers under `pkg/controller/...`.
+
+The Manager can restrict the namespace that all controllers will watch for resources:
 
 ```Go
-func main() {
-  sdk.Watch("cache.example.com/v1alpha1", "Memcached", "default", 5)
-  sdk.Handle(stub.NewHandler())
-  sdk.Run(context.TODO())
-}
+mgr, err := manager.New(cfg, manager.Options{Namespace: namespace})
 ```
 
-#### Define the Memcached spec and status
-
-Modify the spec and status of the `Memcached` CRD at `pkg/apis/cache/v1alpha1/types.go`:
+By default this will be the namespace that the operator is running in. To watch all namespaces leave the namespace option empty:
 
 ```Go
-type MemcachedSpec struct {
-	// Size is the size of the memcached deployment
-	Size int32 `json:"size"`
-}
-type MemcachedStatus struct {
-	// Nodes are the names of the memcached pods
-	Nodes []string `json:"nodes"`
-}
+mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
 ```
-Update the generated code for the CR:
+
+## Add a new Custom Resource Definition
+
+Add a new Custom Resource Definition (CRD) API called `Memcached`, with APIVersion `cache.example.com/v1apha1` and Kind `Memcached`.
 
 ```sh
-$ cd $GOPATH/src/github.com/example-inc/
-$ cd memcached-operator
-$ operator-sdk generate k8s
+$ operator-sdk add api --api-version=cache.example.com/v1alpha1 --kind=Memcached
+```
+
+This will scaffold the `Memcached` resource API under `pkg/apis/cache/v1alpha1/...`.
+
+## Add a new Controller
+
+Add a new [Controller][controller_go_doc] to the project that will watch and reconcile the `Memcached` resource:
+
+```sh
+$ operator-sdk add controller --api-version=cache.example.com/v1alpha1 --kind=Memcached
 ```
 
-#### Define the Handler
+This will scaffold a new Controller implementation under `pkg/controller/memcached/...`.
+
+For this example replace the generated Controller file `pkg/controller/memcached/memcached_controller.go` with the example [`memcached_controller.go`][memcached_controller] implementation.
```

Comment:
The `memcached_controller` link doesn't exist. This should be pointing to the file that replaces the handler.